docs/restrict-cross-namespace-associations.asciidoc

```diff
@@ -0,0 +1,97 @@
+ifdef::env-github[]
+****
+link:https://www.elastic.co/guide/en/cloud-on-k8s/master/restrict-cross-namespace-associations.html[View this document on the Elastic website]
+****
+endif::[]
+[id="{p}-restrict-cross-namespace-associations"]
+=== Restrict cross namespace resource associations
+
+By default, associations can be created between any resources as long as they are deployed in namespaces managed by the same ECK instance.
+
+This is true, for example, when using the `elasticsearchRef` field to automatically configure and establish a trust relationship between Kibana and Elasticsearch. If a user can create a Kibana instance in a namespace managed by ECK, then this Kibana instance can be associated with any Elasticsearch cluster deployed in another namespace managed by the same ECK instance. Even if the user only has permissions to edit Kibana and not the Elasticsearch resource.
+
+This section describes how you can restrict the associations that can be created between resources managed by ECK.
+
+[float]
+==== Enabling cross namespace association restriction
+
+This feature is disabled by default. To enable it, start the operator with the `--enforce-rbac-on-refs` flag.
+
+NOTE: This feature only provides an access control for resources deployed across two different namespaces. Associations can still be freely created between resources deployed in a same namespace.
+
+Once enabled, associations can be created as long as the `ServiceAccount` specified in the associated resources can execute a `get` request against the referenced Elasticsearch object.
+
+IMPORTANT: ECK unbinds any association that is not allowed. If you have some existing associations, do not enable this feature without creating the required `Roles` and `RoleBindings` as described in the following sections.
+
+[float]
+=== Specify a ServiceAccount in an associated resource
```

Comment:
heading hierarchy is inconsistent, this is higher than the preceding "Enabling cross namespace association restriction" section, probably not intentional

Reply:
I find this heading slightly confusing, even though I think I  know what you mean. But I don't have a good replacement suggestion right now. Maybe "Restricting access to associated resources"? Even though you don't need to specify a service account for that, so maybe just move it to the next section?